clang/test/SemaCXX/warn-unsafe-buffer-usage-array.cpp

```diff
@@ -39,6 +39,23 @@ void constant_idx_unsafe(unsigned idx) {
   buffer[10] = 0;       // expected-note{{used in buffer access here}}
 }
 
+enum FooEnum {
+  A = 0,
+  B = 1,
+  C = 2,
+  D
+};
+
+void constant_enum_safe() {
+  int buffer[FooEnum::D] = { 0, 1, 2 };
+  buffer[C] = 0; // no-warning
```

Comment:
A test for `buffer[A]` (0, no warn) and `buffer[D]` (warn) would be good

Reply:
Good suggestion, fixed :)